lib/rust/flexer/src/automata/pattern.rs

```diff
@@ -91,7 +93,12 @@ impl Pattern {
 
     /// Pattern that triggers when sequence of characters given by `chars` is encountered.
     pub fn all_of(chars:&str) -> Self {
-        chars.chars().fold(Self::never(),|pat,char| pat >> Self::char(char))
+        let char_vec = chars.chars().collect_vec();
+        if let Some((first,rest)) = char_vec.split_first() {
+            rest.iter().fold(Self::char(*first),|pat, char| pat >> Self::char(*char))
+        } else {
+            Pattern::never()
+        }
```

Comment:
I don't think we need to allocate a vector here. We just iterate anyway.

```suggestion
        let mut chars_iter = chars.chars();
        if let Some(first) = chars_iter.next() {
            chars_iter.fold(Self::char(first),|pat, char| pat >> Self::char(char))
        } else {
            Pattern::never()
        }
```